core/field_dropdown.js

```diff
@@ -237,7 +237,7 @@ Blockly.FieldDropdown.prototype.onHide = function() {
   this.dropDownOpen_ = false;
   // Update colour to look selected.
   if (!this.disableColourChange_) {
-    if (this.sourceBlock_.isShadow()) {
+    if (this.sourceBlock_ && this.sourceBlock_.isShadow()) {
```

Comment:
Hey there!

It looks like you want to check for this.sourceBlock_ one line higher:

```
if (!this.disableColourChange_ && this.sourceBlock_) {
  if (this.sourceBlock_isShadow()) ...
```

because sourceBlock_ is also used in the else if, and without a check for its existence.